patsy/test_pickling.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import print_function
+
+import six
+from six.moves import cPickle as pickle
+
+import os
+import shutil
+
+from patsy import EvalFactor
+
+PICKE_TESTCASES_ROOTDIR = os.path.join(os.path.dirname(__file__), '..', 'pickle_testcases')
+
+pickling_testcases = {
+    "evalfactor_simple": EvalFactor("a+b"),
+    }
+
+def test_pickling_same_version_roundtrips():
+    for obj in six.itervalues(pickling_testcases):
+        yield (check_pickling_same_version_roundtrips, obj)
+
+def check_pickling_same_version_roundtrips(obj):
+        assert obj == pickle.loads(pickle.dumps(obj, pickle.HIGHEST_PROTOCOL))
+
+def test_pickling_old_versions_still_work():
+    for (dirpath, dirnames, filenames) in os.walk(PICKE_TESTCASES_ROOTDIR):
+        for fname in filenames:
+            if os.path.splitext(fname)[1] == '.pickle':
+                yield check_pickling_old_versions_still_work, os.path.join(dirpath, fname)
+
+def check_pickling_old_versions_still_work(pickle_filename):
+    testcase_name = os.path.splitext(os.path.basename(pickle_filename))[0]
+    with open(pickle_filename, 'rb') as f:
+        # When adding features to a class, it will happen that there is no
+        # way to make an instance of that version version of that class
+        # equal to any instance of a previous version. How do we handle
+        # that?
```

Comment:
typo: `version version`

And I think the answer here is probably that when we unpickle an old object into a new patsy, if the new patsy has some new features, then we'll have to somehow initialize them for the old object (like fill in zeros or empty strings or whatever), because we need *some* way to represent it in memory. So we can put in a test case here that uses those default field values, and also add a new object that actually uses the new fields in a non-trivial way.